File: FinesModel4/Models/Toma.cs
using System;
using System.Collections.Generic;

namespace FinesModel4.Models;

public partial class Toma
{
    public string Id { get; set; } = null!;

    public DateOnly? FechaToma { get; set; }

    public string? Estado { get; set; }

    public string? Observaciones { get; set; }

    public string? Comentario { get; set; }

    public string TipoMovimiento { get; set; } = null!;

    public string? EstadoContralor { get; set; }

    public DateTime Alta { get; set; }

    public string Curso { get; set; } = null!;

    public string? Docente { get; set; }

    public string? Reemplazo { get; set; }

    public string? PlanillaDocente { get; set; }

    public bool Calificacion { get; set; }

    public bool TemasTratados { get; set; }

    public bool Asistencia { get; set; }

    public bool SinPlanillas { get; set; }

    public bool Confirmada { get; set; }

    public virtual ICollection<AsignacionPlanillaDocente> AsignacionPlanillaDocentes { get; set; } = new List<AsignacionPlanillaDocente>();

    public virtual Curso CursoNavigation { get; set; } = null!;

    public virtual Persona? DocenteNavigation { get; set; }

    public virtual PlanillaDocente? PlanillaDocenteNavigation { get; set; }

    public virtual Persona? ReemplazoNavigation { get; set; }
}

File: src/Fines2/Model/Toma.php
<?php

namespace Fines2\Model;

use SqlOrganize\Sql\Entity;
use Exception;
use DateTime;

class Toma extends Entity
{

    public function __construct()
    {
        $this->_entityName = "toma";
        $this->_db = \App\Context::getFinesDb();
        $this->setDefault();
    }

    /** @var DateTime|null */
    public ?DateTime $alta = null;

    /** @var string|null */
    public ?string $archivo = null;

    /** @var string|null */
    public ?string $comentario = null;

    /** @var string|null */
    public ?string $curso = null;

    /** @var string|null */
    public ?string $docente = null;

    /** @var string|null */
    public ?string $estado = null;

    /** @var string|null */
    public ?string $estado_contralor = null;

    /** @var string|null */
    public ?string $estado_planilla = null;

    /** @var DateTime|null */
    public ?DateTime $fecha_toma = null;

    /** @var string|null */
    public ?string $id = null;

    /** @var string|null */
    public ?string $observaciones = null;

    /** @var string|null */
    public ?string $planilla_docente = null;

    /** @var bool|null */
    public ?bool $reclamo = null;

    /** @var string|null */
    public ?string $reemplazo = null;

    /** @var string|null */
    public ?string $tipo_movimiento = null;

    /** @var Curso|null (fk toma.curso _m:o curso.id) */
    public ?Curso_ $curso_ = null;

    /** @var Persona|null (fk toma.docente _m:o persona.id) */
    public ?Persona_ $docente_ = null;

    /** @var PlanillaDocente|null (fk toma.planilla_docente _m:o planilla_docente.id) */
    public ?PlanillaDocente_ $planilla_docente_ = null;

    /** @var Persona|null (fk toma.reemplazo _m:o persona.id) */
    public ?Persona_ $reemplazo_ = null;

    /** @var int|null */
    public ?int $AsignacionPlanillaDocente_Count = null;

    /** @var AsignacionPlanillaDocente[] (ref asignacion_planilla_docente.toma _m:o toma.id) */
    public array $AsignacionPlanillaDocente_ = [];

}
